src/sentry/incidents/utils/process_update_helpers.py

```diff
@@ -61,3 +72,100 @@ def get_aggregation_value_helper(subscription_update: QuerySubscriptionUpdate) -
+    if query_type == SnubaQuery.Type.PERFORMANCE and dataset == Dataset.EventsAnalyticsPlatform:
+        try:
+            rpc_time_series_request = entity_subscription.build_rpc_request(
+                query=snuba_query.query,
+                project_ids=project_ids,
+                environment=snuba_query.environment,
+                params={
+                    "organization_id": organization_id,
+                    "project_id": project_ids,
+                },
+                referrer="subscription_processor.comparison_query",
+            )
+
+            rpc_time_series_request = add_start_end_conditions(rpc_time_series_request, start, end)
+
+            rpc_response = snuba_rpc.timeseries_rpc([rpc_time_series_request])[0]
+            if len(rpc_response.result_timeseries):
+                comparison_aggregate = rpc_response.result_timeseries[0].data_points[0].data
+
+        except Exception:
+            logger.exception(
+                "Failed to run RPC comparison query",
+                extra={
+                    "alert_rule_id": alert_rule_id,
+                    "subscription_id": subscription_update.get("subscription_id"),
+                    "organization_id": organization_id,
+                },
+            )
+            return None
+
+    else:
+        try:
+            # TODO: determine whether we need to include the subscription query_extra here
+            query_builder = entity_subscription.build_query_builder(
+                query=snuba_query.query,
+                project_ids=project_ids,
+                environment=snuba_query.environment,
+                params={
+                    "organization_id": organization_id,
+                    "project_id": project_ids,
+                    "start": start,
+                    "end": end,
+                },
+            )
+            time_col = ENTITY_TIME_COLUMNS[get_entity_key_from_query_builder(query_builder)]
+            query_builder.add_conditions(
+                [
+                    Condition(Column(time_col), Op.GTE, start),
+                    Condition(Column(time_col), Op.LT, end),
+                ]
+            )
+            query_builder.limit = Limit(1)
+            results = query_builder.run_query(referrer="subscription_processor.comparison_query")
+            comparison_aggregate = list(results["data"][0].values())[0]
+
+        except Exception:
+            logger.exception(
+                "Failed to run comparison query",
+                extra={
+                    "alert_rule_id": alert_rule_id,
+                    "subscription_id": subscription_update.get("subscription_id"),
+                    "organization_id": organization_id,
+                },
+            )
+            return None
+
+    if not comparison_aggregate:
+        metrics.incr("incidents.alert_rules.skipping_update_comparison_value_invalid")
+        return None
+
+    return (aggregation_value / comparison_aggregate) * 100
```

Comment:
🤔 could these be decomposed into smaller methods? it seems like the `if` statements would be good breakdowns for smaller methods, then have this call into those. (mostly that just helps the reading this code in the future.)

Reply:
Would you mind if I did that in a follow up? I was trying to make no changes to the function in this PR beyond moving it.

Reply:
sg to do as a separate follow, otherwise lgtm!